lib/collection/src/shards/transfer/driver.rs

```diff
@@ -43,7 +44,10 @@ pub async fn transfer_shard(
     snapshots_path: &Path,
     temp_dir: &Path,
 ) -> CollectionResult<bool> {
-    let shard_id = transfer_config.shard_id;
+    // The remote might target a different shard ID depending on the shard transfer type
+    let shard_id = transfer_config
+        .to_shard_id
+        .unwrap_or(transfer_config.shard_id);
```

Comment:
This does not seem right.

- we initialize `shard_id` to `to_shard_id`, and initialize `remote_shard` with it, which is correct...
- ...but then we pass the same `shard_id` (which is initialized to `to_shard_id`) into `transfer_resharding_stream_records`, which treats `shard_id` as a *source* shard id
- e.g., `transfer_resharding_stream_records` expects `shard_id` to be a local shard that has to be proxified, but `shard_id` is initialized to `to_shard_id` in this case, so we try to proxify a remote shard to itself :D

Reply:
Fixed in 782d78b99a3a7c739b6149015b1100621e621c6a